### DISCUSSION TITLE
Property is already defined

### DISCUSSION THREAD

**Discussion by hudkinsnoah** [JSBSim-Team/jsbsim#854] [Support requests]
I am pretty new to JSBSim and am taking over a project from another team member who is no longer around to answer questions. I haven't been able to find any answers elsewhere about a warning "Property __ is already defined. There are about 19 properties that notify me on startup of the application. 

JSBSim has been integrated into other internal software and has worked great, until working with scripting. Unbind is called on an instance of FDMExec and then loadScript is called. The call never returns and the application exits. This is a snippet from the console window for the messages that I am receiving: 

```
> In file systems/GNCUtilities.xml: line 69
>       Property guidance/specified-heading-rad is already defined.
> 
> In file systems/GNCUtilities.xml: line 70
>       Property guidance/heading-selector-switch is already defined.
> 
> In file aircraft/c310/c310ap.xml: line 28
>       Property ap/attitude_hold is already defined.
> 
> In file aircraft/c310/c310ap.xml: line 29
>       Property ap/altitude_hold is already defined.
> 
> In file aircraft/c310/c310ap.xml: line 30
>       Property ap/heading_hold is already defined.
> 
> In file aircraft/c310/c310ap.xml: line 31
>       Property ap/yaw_damper is already defined.
```

The only place where each of these properties are defined that I can see if at the location it gives. I am not sure that the application exiting and these messages are related, but figured I would mention it. Any help would be appreciated, and I am happy to provide as much information as I can. 

Thanks!

**Comment by hudkinsnoah** [comment]
I changed the configuration to instead use c172x scripts and it has the same issue, however it does create a csv output file. It only has one line though and the program again exits. Any help would still be appreciated

**Comment by ermarch** [comment]
Looking at the code I have the sneaking suspicion that the use of the (now) reserved name 'override' as a variable name in input_output/FGPropertyReader.h and input_output/FGPropertyReader.cpp might be causing this. You could try to rename it to something like props_override.

Update:
I now see that math/FGModelFunctions.cpp calls the LocalProperties Load function with override set to false.
This is the only place in the code where this function is called with override set to false. All other occurrences set it to true.

**Reply by hudkinsnoah** [reply]
Changing to true fixed it! Thank you!

**Reply by ermarch** [reply]
Now is the question, why was it set to false for that particular function and could it be set to true instead.

**Reply by bcoconni** [reply]
> Now is the question, why was it set to false for that particular function and could it be set to true instead.

It is set to `false` because JSBSim considers illegal to declare a property that has already been defined elsewhere (more details in the PR #864).

There is one exception however where the variable `override` is set to true and this occurs in 
https://github.com/JSBSim-Team/jsbsim/blob/e1af7cea6ce6dfc35cad99a005df6cce66a0b328/src/input_output/FGScript.cpp#L213-L217

as we want the scripts being able to override the default/initial values set in the XML definitions files.

**Comment by bcoconni** [comment]
@hudkinsnoah these messages are harmless and are merely warning messages about encountering XML statements such as `<property> guidance/specified-heading-rad </property>` while the property `guidance/specified-heading-rad` has already been defined elsewhere, knowing that an XML statement such as below defines the property `guidance/specified-heading-rad`.

```xml
      <test value="guidance/specified-heading-rad">
        guidance/heading-selector-switch eq 1
      </test>
```

Is there a chance that you could locate where the application hangs/crashes and tell us where in JSBSim that occurs ?
